src/VisualStudio/TestUtilities/Remoting/RemotingHelper.cs

```diff
@@ -1,86 +1,180 @@
 
-        public static InteractiveWindowWrapper CreateCSharpInteractiveWindowWrapper()
-        {
-            var componentModel = (IComponentModel)(ServiceProvider.GlobalProvider.GetService(typeof(SComponentModel)));
-            var vsInteractiveWindowProvider = componentModel.GetService<CSharpVsInteractiveWindowProvider>();
-            var vsInteractiveWindow = ExecuteOnUIThread(() => vsInteractiveWindowProvider.Open(0, true));
-            return new InteractiveWindowWrapper(vsInteractiveWindow.InteractiveWindow);
-        }
+        private static readonly string[] SupportedLanguages = new string[] { LanguageNames.CSharp, LanguageNames.VisualBasic };
 
-        public static string GetActiveTextViewContents()
+        public static string ActiveTextViewContents
         {
-            var textViewHost = GetActiveTextViewHost();
-            return ExecuteOnUIThread(textViewHost.TextView.TextSnapshot.GetText);
+            get
+            {
+                return InvokeOnUIThread(ActiveTextView.TextSnapshot.GetText);
+            }
+
+            set
+            {
+                InvokeOnUIThread(() => {
+                    var textSnapshot = ActiveTextView.TextSnapshot;
+                    var replacementSpan = new SnapshotSpan(textSnapshot, 0, textSnapshot.Length);
+                    ActiveTextView.TextBuffer.Replace(replacementSpan, value);
+                });
+            }
         }
 
-        public static void SetActiveTextViewContents(string text)
+        public static ReadOnlyCollection<ICompletionSession> ActiveTextViewCompletionSessions => CompletionBroker.GetSessions(ActiveTextView);
+
+        public static IComponentModel ComponentModel => GetGlobalService<IComponentModel>(typeof(SComponentModel));
+
+        public static IInteractiveWindow CSharpInteractiveWindow => CSharpVsInteractiveWindow.InteractiveWindow;
+
+        public static IOptionService OptionService => VisualStudioWorkspaceServices.GetService<IOptionService>();
+
+        public static VisualStudioWorkspace VisualStudioWorkspace => ComponentModel.GetService<VisualStudioWorkspace>();
+
+        private static ITextView ActiveTextView => ActiveTextViewHost.TextView;
+
+        private static IWpfTextViewHost ActiveTextViewHost
         {
-            var textViewHost = GetActiveTextViewHost();
-            ExecuteOnUIThread(() => {
-                var textView = textViewHost.TextView;
-                var textSnapshot = textView.TextSnapshot;
-                var replacementSpan = new SnapshotSpan(textSnapshot, 0, textSnapshot.Length);
-                textView.TextBuffer.Replace(replacementSpan, text);
-            });
+            get
+            {
+                // The active text view might not have finished composing yet, waiting for the application to 'idle'
+                // means that it is done pumping messages (including WM_PAINT) and the window should have finished composing
```

Comment:
I'm not aware of any laziness in the system that depends upon message pumping like this to query through IVsUserData. Can you point to the specific mechanism in the VS codebase (i.e. the actual line of code) that requires this?